command/connect/envoy/envoy.go

```diff
@@ -659,7 +659,7 @@ func (c *cmd) xdsAddress() (GRPC, error) {
 	if addr == "" {
 		port, protocol, err := c.lookupXDSPort()
 		if err != nil {
-			c.UI.Error(fmt.Sprintf("Error connecting to Consul agent: %s", err))
+			return g, fmt.Errorf("Could not look up xDS port: %w", err)
```

Comment:
I'm unsure if the return was missed here or we intentionally went on to default to `8502`. It looks safe to error here since if the call to `Agent.Self` doesn't yield any port, we probably don't want the command to continue.

Reply:
It can't error out because typically `agent:read` is not granted to sidecars, which is required for `/v1/agent/self`.  The existing code did the agent self lookup as an _UX optimization_.

Instead you could pick whichever port we already calculated and do a super simple `net.Dial` with a short timeout and see if the port is even open instead, which wouldn't even need ACL tokens.

Reply:
Yeah, I'm not sure why it never errored-out before. I assume it was to enable some backwards compatibility with old APIs that didn't include the port. I don't honestly know the history on this particular area of code, aside from the recent changes I made. I'll have to think about it quite a bit, because there might be some situation where the API is not available, but gRPC is.

Reply:
Nevermind. RB already answered the question.

Reply:
@rboyer I ended up logging and defaulting only if it's an ACL perm denied and otherwise erroring out.